dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java

```diff
@@ -123,16 +124,29 @@ public Set<String> getMissingFeaturesFromRepo(List<String> mavenCoords, MavenRep
         return missingCoords;
     }
 
-    public void synthesizeAndDownloadFeatures(List<String> mavenCoords, String dLocation, MavenRepository repository) throws InstallException {
+    public void synthesizeAndDownloadFeatures(List<String> mavenCoords, String dLocation, MavenRepository repository, String verifyOptiion,
+                                              boolean downloadAll) throws InstallException {
         final List<Future<?>> futures = new ArrayList<>();
-        // we have downloaded mavenCoords.length * 2 (esa and pom) amount of features.
-        double individualSize = progressBar.getMethodIncrement("downloadArtifacts") / (2 * mavenCoords.size());
+        double individualSize = 0;
         info(Messages.INSTALL_KERNEL_MESSAGES.getMessage("MSG_BEGINNING_DOWNLOAD_FEATURES"));
         for (String coords : mavenCoords) {
-            Future<?> future1 = submitDownloadRequest(coords, "esa", dLocation, repository);
-            futures.add(future1);
-            Future<?> future2 = submitDownloadRequest(coords, "pom", dLocation, repository);
-            futures.add(future2);
+            Future<?> future;
+            if (downloadAll) {
+                // we have downloaded mavenCoords.length * 2 (esa and pom) amount of features.
+                individualSize = progressBar.getMethodIncrement("downloadArtifacts") / (2 * mavenCoords.size());
+                future = submitDownloadRequest(coords, "esa", dLocation, repository);
+                futures.add(future);
+                future = submitDownloadRequest(coords, "pom", dLocation, repository);
+                futures.add(future);
+            }
+            if (verifyOptiion != null) {
+                // we have downloaded mavenCoords.length * 4 (esa,asc and poms) amount of features.
+                individualSize = progressBar.getMethodIncrement("downloadArtifacts") / (4 * mavenCoords.size());
```

Comment:
if downloadAll is true and verifyOption != null, would we increment the progress bar too far?